tuning/sft_trainer.py

```diff
@@ -200,6 +200,8 @@ def train(
         ),
         cache_dir=train_args.cache_dir,
         use_fast=True,
+        legacy=False,
```

Comment:
trick question I dont know the answer to!

> This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565 - if you loaded a llama tokenizer from a GGUF file you can ignore this message.

Do we want this "new behavior"? What is the new vs old behavior?

Reply:
So it seems like it defaults to [legacy=None, then legacy=True](https://github.com/huggingface/transformers/pull/24565#issuecomment-1680314450) but gives the warning. If we hardcode [legacy=True](https://github.com/huggingface/transformers/pull/24565#issuecomment-1680587061), then it will use the behavior we have been using in the past. It seems that [llama models use legacy=False](https://github.com/huggingface/transformers/pull/24565#issuecomment-1661679624) in their tokenizers. I think for the time being I will set legacy=True so everything remains as it has been, but we could do some quality testing in the future for legacy=False.